k4EDM4hep2LcioConv/include/k4EDM4hep2LcioConv/k4Lcio2EDM4hepConv.ipp

```diff
@@ -2,44 +2,59 @@
 
 #include <UTIL/PIDHandler.h>
 
+#include "podio/Frame.h"
+
 #include <edm4hep/ParticleIDCollection.h>
 #include <edm4hep/RecDqdxCollection.h>
 
+#include "k4FWCore/MetadataUtils.h"
+
 namespace LCIO2EDM4hepConv {
 template <typename LCIOType>
-void convertObjectParameters(LCIOType* lcioobj, podio::Frame& event) {
+void convertObjectParameters(LCIOType* lcioobj, std::optional<std::reference_wrapper<podio::Frame>>& event) {
   const auto& params = lcioobj->getParameters();
+
+  // When using the PodioDataSvc there is always access to the event frame
+  // but when using IOSvc there is not a frame
+  auto putParamFun = [&](const std::string& key, const auto& value) {
+    if (event) {
+      event->get().putParameter(key, value);
+    } else {
+      k4FWCore::putParameter(key, value);
+    }
+  };
```

Comment:
Can we make this a template parameter to the `convertObjecParameters`? Then we could externalize the k4FWCore dependency and simply define this in the wrapper, where we already have that dependency. I suppose the main issue is that it would need to be a templated function because it has to work for several datatypes?